CCPAPI/sip/CMakeLists.txt

```diff
@@ -0,0 +1,149 @@
+#                                                                        #
+##########################################################################
+
+cmake_minimum_required(VERSION 3.0)
+cmake_policy(SET CMP0071 NEW)
+message(STATUS "SIP processing")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../cmake" )
+
+# --- Python detection: if several versions installed, select one with all the required packages installed
+
+set(PYTHON_PREFERED_VERSION "3.6" CACHE STRING "Select a Python version: it should be installed with all packages required for CloudCompare (pyqt, sip, numpy)")
+set_property(CACHE PYTHON_PREFERED_VERSION PROPERTY STRINGS "3.6" "3.7" "3.8")
+message(STATUS "python prefered version: " ${PYTHON_PREFERED_VERSION})
+set(Python_ADDITIONAL_VERSIONS ${PYTHON_PREFERED_VERSION}) # priority to this version 
+find_package(PythonInterp)
+find_package(PythonLibs)
+message(STATUS "python version: " ${PYTHON_VERSION_STRING})
+message(STATUS "check match for ${PYTHON_PREFERED_VERSION}.x")
+if(NOT ${PYTHON_VERSION_STRING} MATCHES "${PYTHON_PREFERED_VERSION}\\..")
+    message(SEND_ERROR "no Python found matching ${PYTHON_PREFERED_VERSION}.x")
+else()
+    message(STATUS "Python found ${PYTHON_VERSION_STRING} matches prefered version ${PYTHON_PREFERED_VERSION}.x")
+    message(STATUS "PYTHON_LIBRARIES:" ${PYTHON_LIBRARIES})
+endif()
+
+# --- PyQt Sip detection
+
+find_package(PYQTSIP REQUIRED)  # should come after Python
+
+# --- ----------------------------------------------------------
+# --- target PyQt sip Python library: Python module cloudCompare
+
+project( cloudCompare VERSION 0.1 )
+
+# --- sources / sip wrappings
+
+set(SIP_MAIN_SOURCE cloudCompare.sip)
+set(SIP_ADD_SOURCES
+    ccPApi.sip
+    ccPointCloud.sip
+    ScalarField.sip
+    )
+set(SIP_ALL_SOURCES ${SIP_MAIN_SOURCE} ${SIP_ADD_SOURCES})
+message(STATUS "SIP_ALL_SOURCES: " "${SIP_ALL_SOURCES}")
+
+# TODO: names of generated sip files follow some rules based on module name, class name, namespace etc. Not obvious to compute...
+
+set(SIP_MAIN_GENSRC ${CMAKE_CURRENT_BINARY_DIR}/sipcloudComparecmodule.cc)
+set(SIP_ADD_GENSRC
+    ${CMAKE_CURRENT_BINARY_DIR}/sipcloudCompareccPointCloud.cc
+    ${CMAKE_CURRENT_BINARY_DIR}/sipcloudCompareQList0101ccPointCloud.cc
+    ${CMAKE_CURRENT_BINARY_DIR}/sipcloudCompareCCLib.cc
+    ${CMAKE_CURRENT_BINARY_DIR}/sipcloudCompareCCLibScalarField.cc
+   )
+set(SIP_ALL_GENSRC ${SIP_MAIN_GENSRC} ${SIP_ADD_GENSRC})
+
+message(STATUS "SIP_ALL_GENSRC: " "${SIP_ALL_GENSRC}")
+
+set_source_files_properties( ${SIP_ALL_GENSRC} PROPERTIES GENERATED TRUE )
+
+set(SIP_OPTIONS -s .cc -c . ${PYQT_SIPFLAGS})
+message(STATUS "SIP_OPTIONS: " "${SIP_OPTIONS}")
+
+add_custom_command(
+    OUTPUT ${SIP_ALL_GENSRC}
+    COMMAND ${SIP_EXECUTABLE} ${SIP_OPTIONS} ${CMAKE_CURRENT_SOURCE_DIR}/${SIP_MAIN_SOURCE}
+    DEPENDS ${SIP_ALL_SOURCES}
+    )
+
+# --- sources / to compile
+
+add_library( cloudCompare SHARED ${SIP_ALL_GENSRC} )
```

Comment:
Suggest `add_library` move above SIP generation and then use `target_sources(  cloudCompare PRIVATE ${SIP_ALL_GENSRC} )`